### DISCUSSION TITLE
Next 13 static export is fetching twice

### DISCUSSION THREAD

**Discussion by nareshbhatia** [vercel/next.js#48935] [Help] [answered]
### Summary

I am trying to run Next 13 as a pure SPA using [static export](https://beta.nextjs.org/docs/configuring/static-export). I am finding that my client component is fetching twice - even in a production build (the console log in the `getMovies()` function is printed twice. Why is this happening? Am I missing something?

Full code shown below. The repo is [here](https://github.com/nareshbhatia/movie-magic-2/blob/main/apps/movie-magic-nextjs-static-build/src/app/top-10-movies/Top10MoviesList.tsx).

### Additional information

```js
'use client';

import Image from 'next/image';

async function getMovies() {
  const API_URL = process.env.NEXT_PUBLIC_API_URL;
  console.log('----> API_URL', API_URL);
  const resMovies = await fetch(`${API_URL}/top-10-movies`);
  // returns a promise that resolves to movies in JSON format
  return resMovies.json();
}

export async function Top10MoviesList() {
  const movies = await getMovies();
  return (
    <ul className="flex-none space-y-6">
      {movies.map((movie: any) => (
        <li key={movie.id}>
          <div className="bg-surface border border-transparent rounded-xl min-h-[160px] flex overflow-hidden">
            <div className="flex-none w-32 relative">
              <Image
                src={movie.photoUrl}
                alt={movie.name}
                fill={true}
                sizes="256px"
                className="object-cover"
              />
            </div>
            <div className="flex-auto p-4">
              <div className="flex">
                <h1 className="flex-auto text-base leading-5 font-semibold">
                  {movie.name}
                </h1>
                <div className="ml-2 flex text-base leading-5 font-semibold">
                  {movie.rating.toFixed(1)}
                </div>
              </div>
              <div className="w-full flex-none text-sm font-medium">
                {movie.year}
              </div>
              <div className="mt-2 text-sm">{movie.logline}</div>
            </div>
          </div>
        </li>
      ))}
    </ul>
  );
}
```

### Example

https://github.com/nareshbhatia/movie-magic-2/blob/main/apps/movie-magic-nextjs-static-build/src/app/top-10-movies/Top10MoviesList.tsx

**Comment by nareshbhatia** [comment]
Some added context: my team currently uses CRA to build internal apps. Initial page load performance is not critical - users are logged into the apps all day. The team is very happy to throw the static builds at any web server - it's really simple. I am trying to move them from CRA to Next, the first step being move to a Next static build. However, an issue like the one above causes concerns. Would love to get a good explanation/resolution.

P.S. One benefit that I see (even with Next static builds) is the automatic code splitting of the deployed bundle. This should improve initial page load performance anyway.

CC: @leerob

**Reply by leerob** [reply, marked as answer]
Happy to help – you can't use `async` / `await` currently with client components. We have since merged an ESLint error to make this more clear. In the meantime, to do data fetching, you can either keep a server component (this will run during the build when using a static export) or do data fetching on the client side (similar to CRA) with `useEffect` or similar React hook libraries.

**Reply by nareshbhatia** [reply]
Thank you, @leerob. Such a relief to know this :-)
